analysis.js

```diff
@@ -2,70 +2,166 @@ $("#chart-weeklyDatePicker").datetimepicker({
   format: 'YYYY-MM-DD'
 });
 
-$("#chart-weeklyDatePicker").on("dp.change", function() {
-  var value = $("#chart-weeklyDatePicker").val();
-  var firstDate = moment(value, "YYYY-MM-DD").day(0).format("YYYY-MM-DD");
-  var lastDate = moment(value, "YYYY-MM-DD").day(6).format("YYYY-MM-DD");
-
-  $('#analysis-name').html('<h3 class="text-center">資料載入中...</h3>');
-  $("#chart-weeklyDatePicker").val(firstDate + "~" + lastDate);
+$("#chart-weeklyDatePickerStart").datetimepicker({
+  format: 'YYYY-MM-DD'
+});
 
-  window.fetchWeek($("#chart-weeklyDatePicker").val(), function() {
-    updateTownFormAndTitle();
-  });
+$("#chart-weeklyDatePickerEnd").datetimepicker({
+  format: 'YYYY-MM-DD'
 });
 
+function compareDate(firstdate, enddate, start, end)
+{
+  if(start.length == 0) {
+    var aidDate = moment(end, "YYYY-MM-DD").subtract(6, 'days').format("YYYY-MM-DD");
+    $("#chart-weeklyDatePickerStart").val(aidDate);
+    return false;
+  } else if(end.length == 0) {
+    var aidDate = moment(start, "YYYY-MM-DD").add(6, 'days').format("YYYY-MM-DD");
+    $("#chart-weeklyDatePickerEnd").val(aidDate);
+    return false;
+  }
+  if(start.length != 0  && firstdate <= enddate) {
+    var date1 = moment(start, "YYYY-MM-DD");
+    var date2 = moment(end, "YYYY-MM-DD");
+    var dayDiff = date2.diff(date1, 'days');
+    console.log(dayDiff);
+    if(dayDiff > 30 || dayDiff < 7) {
```

Comment:
The rule should be displayed to users, otherwise they may confuse why there is no response